src/agent/src/command_handler_utils.cpp

```diff
@@ -66,47 +66,50 @@ namespace
     }
 } // namespace
 
-boost::asio::awaitable<module_command::CommandExecutionResult>
+boost::asio::awaitable<std::optional<module_command::CommandExecutionResult>>
 DispatchCommand(module_command::CommandEntry commandEntry,
-                std::function<boost::asio::awaitable<module_command::CommandExecutionResult>(
-                    std::string command, nlohmann::json parameters)> executeFunction,
-                std::shared_ptr<IMultiTypeQueue> messageQueue)
+                std::function<boost::asio::awaitable<void>(module_command::CommandEntry commandEntry)> enqueueFunction)
 {
     using namespace boost::asio::experimental::awaitable_operators;
 
     LogInfo("Dispatching command {}({})", commandEntry.Command, commandEntry.Module);
 
-    auto result = std::make_shared<module_command::CommandExecutionResult>();
-
-    co_await (TimerTask(result) || ExecuteCommandTask(executeFunction, commandEntry, result));
-
-    commandEntry.ExecutionResult.ErrorCode = result->ErrorCode;
-    commandEntry.ExecutionResult.Message = result->Message;
-
-    ReportCommandResult(commandEntry, messageQueue);
-
-    co_return *result;
+    co_await enqueueFunction(commandEntry);
+    co_return std::nullopt;
 }
 
-boost::asio::awaitable<module_command::CommandExecutionResult>
+boost::asio::awaitable<std::optional<module_command::CommandExecutionResult>>
 DispatchCommand(module_command::CommandEntry commandEntry,
                 std::shared_ptr<ModuleWrapper> module,
                 std::shared_ptr<IMultiTypeQueue> messageQueue)
 {
+    using namespace boost::asio::experimental::awaitable_operators;
+
     if (!module)
     {
         LogError("Error dispatching command: module {} not found", commandEntry.Module);
         co_return module_command::CommandExecutionResult {module_command::Status::FAILURE, "Module not found"};
     }
 
+    LogInfo("Dispatching command {}({})", commandEntry.Command, commandEntry.Module);
+
+    auto result = std::make_shared<module_command::CommandExecutionResult>();
+
     auto moduleExecuteFunction =
         [module](const std::string& command,
                  const nlohmann::json& parameters) -> boost::asio::awaitable<module_command::CommandExecutionResult>
     {
         return module->ExecuteCommand(command, parameters);
     };
 
-    co_return co_await DispatchCommand(commandEntry, moduleExecuteFunction, messageQueue);
+    co_await (TimerTask(result) || ExecuteCommandTask(moduleExecuteFunction, commandEntry, result));
+
+    commandEntry.ExecutionResult.ErrorCode = result->ErrorCode;
+    commandEntry.ExecutionResult.Message = result->Message;
+
+    ReportCommandResult(commandEntry, messageQueue);
```

Comment:
To keep code better organized, I think this should be called [here](https://github.com/wazuh/wazuh-agent/pull/499/files#diff-b1ac96b521cefa542c6738817b12ae65897f4bfc38df66b03553ca873c599a85R102).

Reply:
Done!